src/carrier.rs

```diff
@@ -13,34 +13,49 @@
 // limitations under the License.
 
 use std::fmt;
-use std::ops::Deref;
+
+use crate::ParseError;
 
 /// A phone number carrier.
+/// see: https://en.wikipedia.org/wiki/Mobile_country_code#National_operators
 #[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Hash, Debug)]
-pub struct Carrier(pub(crate) String);
-
-impl<T: Into<String>> From<T> for Carrier {
-    fn from(value: T) -> Carrier {
-        Carrier(value.into())
-    }
+pub struct Carrier {
+    pub mcc: u16, // always 3 digits
+    pub mnc: u16, // 2 or 3 digits
 }
```

Comment:
Is this one of the rare occasions where we should prefer getters to fields? We might change the representation over time, e.g. by compressing `mcc` and `mnc` in a single u16.

Additionally, MNC 001 is not the same as MNC 01, so I'm not sure this representation is the safest.

Reply:
Fair point, I guess we could make it `&str` depending on the lifetime of `PhoneNumber`. I'll rework the PR ASAP.